code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1324,3 +1324,232 @@
+// ***************************************
+// *********** Tail Stab
+// ***************************************
+//totally not stolen from punch code
+/datum/action/ability/activable/xeno/tail_stab
+	name = "Tail Stab"
+	action_icon_state = "tail_attack"
+	desc = "Strike a target within two tiles with a sharp tail for armor-piercing damage, stagger and slowdown. Deals double AP, damage, stagger and slowdown to grappled targets, structures and machinery."
+	ability_cost = 30
+	cooldown_duration = 10 SECONDS
+	use_state_flags = ABILITY_USE_AGILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_TAIL_STAB,
+	)
+	target_flags = ABILITY_MOB_TARGET|ABILITY_MOB_TARGET
+	var/range = 2
+	var/punch_description = "swift tail-jab!"
+
+/datum/action/ability/activable/xeno/tail_stab/on_cooldown_finish()
+	var/mob/living/carbon/xenomorph/X = owner
+	to_chat(X, span_xenodanger("We gather our strength to stab again."))
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+/datum/action/ability/activable/xeno/tail_stab/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+	if(!line_of_sight(owner, A, range))
+		if(!silent)
+			to_chat(owner, span_xenodanger("Our target must be closer!"))
+		return FALSE
+
+	if(A.resistance_flags & (INDESTRUCTIBLE|CRUSHER_IMMUNE)) //no bolting down indestructible airlocks
+		if(!silent)
+			to_chat(owner, span_xenodanger("We cannot damage this target!"))
+		return FALSE
+
+	if(isxeno(A))
+		if(!silent)
+			owner.visible_message(span_xenowarning("\The [owner] swipes their tail through the air!"), span_xenowarning("We swipe our tail through the air!"))
+			add_cooldown(1 SECONDS)
+			playsound(owner, "alien_tail_swipe", 50, TRUE)
+			owner.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
+		return FALSE
+
+	if(!isliving(A) && !isstructure(A) && !ismachinery(A) && !isvehicle(A))
+		if(!silent)
+			owner.visible_message(span_xenowarning("\The [owner] swipes their tail through the air!"), span_xenowarning("We swipe our tail through the air!"))
+			add_cooldown(1 SECONDS)
+			playsound(owner, "alien_tail_swipe", 50, TRUE)
+			owner.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
+		return FALSE
+
+	if(isliving(A))
+		var/mob/living/L = A
+		if(L.stat == DEAD)
+			if(!silent)
+				to_chat(owner, span_xenodanger("We don't care about the dead!"))
+			return FALSE
+
+/datum/action/ability/activable/xeno/tail_stab/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/damage = X.xeno_caste.melee_damage * X.xeno_melee_damage_modifier
+	var/target_zone = check_zone(X.zone_selected)
+
+	if(!A.tail_stab_act(X, damage, target_zone))
+		return fail_activate()
+
+	succeed_activate()
+	if(istype(A, /obj/machinery/light))
+		add_cooldown(1 SECONDS)
+	else
+		add_cooldown()
+
+/atom/proc/tail_stab_act(mob/living/carbon/xenomorph/X, damage, target_zone, push = TRUE, punch_description = "swift tail-stab!", stagger_stacks = 2, slowdown_stacks = 2)
```

Comment:
single-letter var
Also, why are you calling a proc that only returns TRUE?
It's never gonna return anything else since there's no code in here.

Reply:
cause it says atom dont have that proc and crashes it seem.

Reply:
The point remains, why are you calling a proc that only returns TRUE? You can simply not call a proc at all...

Reply:
this is the same way on warrior punch is and i based this off on and it dont seem to work without it and i dont know how i would make it work another way, since removing that seem to break it since tail_stab_act isnt called. and if i only remove the proc it just says atom does not have that proc. I dont really know how to reliably change that, plus woyer punch uses it so its fine i guess.